app/javascript/react/components/Map/Map.tsx

```diff
@@ -197,79 +188,153 @@ const Map = () => {
       }, 10);
       return () => clearInterval(intervalId);
     }
-  }, [currentUserSettings, mapInstance]);
+  }, [mapInstance, previousUserSettings]);
+
+  useEffect(() => {
+    if (streamId && currentUserSettings === UserSettings.ModalView) {
+      fixedSessionTypeSelected
+        ? dispatch(fetchFixedStreamById(streamId))
+        : dispatch(fetchMobileStreamById(streamId));
+    }
+  }, [streamId, currentUserSettings, fixedSessionTypeSelected]);
 
   // Callbacks
-  const onIdle = useCallback(
+  const handleMapIdle = useCallback(
     (event: MapEvent) => {
-      if (currentUserSettings === UserSettings.MapView) {
-        const map = event.map;
-        if (!mapInstance) {
-          setMapInstance(map);
-          map.setOptions({
-            clickableIcons: false,
-          });
+      const map = event.map;
+      if (!mapInstance) {
+        setMapInstance(map);
+        map.setOptions({
+          clickableIcons: false,
+        });
+      }
+
+      if (isFirstRender.current) {
+        if (currentUserSettings === UserSettings.MapView) {
+          newSearchParams.set(UrlParamsTypes.sessionType, sessionType);
+          map.setCenter(currentCenter);
+          map.setZoom(currentZoom);
         }
-        const bounds = map?.getBounds();
-        if (!bounds) {
-          return;
+      } else {
+        if (currentUserSettings === UserSettings.MapView) {
+          const currentCenter = JSON.stringify(
+            map.getCenter()?.toJSON() || previousCenter
+          );
+          const currentZoom = (map.getZoom() || previousZoom).toString();
+          const bounds = map?.getBounds();
+          if (!bounds) {
+            return;
+          }
+          const north = bounds.getNorthEast().lat();
+          const south = bounds.getSouthWest().lat();
+          const east = bounds.getNorthEast().lng();
+          const west = bounds.getSouthWest().lng();
+          newSearchParams.set(UrlParamsTypes.boundEast, east.toString());
+          newSearchParams.set(UrlParamsTypes.boundNorth, north.toString());
+          newSearchParams.set(UrlParamsTypes.boundSouth, south.toString());
+          newSearchParams.set(UrlParamsTypes.boundWest, west.toString());
+          newSearchParams.set(UrlParamsTypes.currentCenter, currentCenter);
+          newSearchParams.set(UrlParamsTypes.currentZoom, currentZoom);
+          navigate(`?${newSearchParams.toString()}`);
         }
-        const north = bounds.getNorthEast().lat();
-        const south = bounds.getSouthWest().lat();
-        const east = bounds.getNorthEast().lng();
-        const west = bounds.getSouthWest().lng();
-        setMapBounds({ north, south, east, west });
       }
     },
-    [mapInstance, currentUserSettings]
+    [currentUserSettings, debouncedUpdateURL, mapInstance, searchParams]
   );
 
   // Handlers;
-  const handleMarkerClick = (streamId: number | null, id: number | null) => {
+  const handleMarkerClick = (
+    selectedStreamId: number | null,
```

Comment:
I renamed this params from `streamId` to `selectedStreamId` to be able to use `streamId` in the URL 💪